### ISSUE TITLE
pyproj crs equality comparison became much slower in 3.0

### ISSUE THREAD

**Issue by mityakrylov** [pyproj4/pyproj#880]
#### Code Sample, a copy-pastable example if possible

```python
WGS = CRS4326 = pyproj.CRS('epsg:4326')

WGS == CRS4326
```

#### Problem description

Test for equality takes about ~10ms on pyproj==3.0.1 and ~10µs on pyproj==2.6.0. I measured in jupyter notebook using timeit cell magic:

```python
%%time
WGS == CRS4326
```

I noticed that when function that was applied to thousands of geometries became much slower after pyproj upgrade. And the main reason was increased time for crs comparison. 

Is this expected behavior in pyproj==3.0.1? If yes, what is better way to test crs for equality? Maybe something like this:
```
WGS.srs == CRS4326.srs
```
?

#### Environment Information

for pyproj==3.0.1:
```
pyproj info:
    pyproj: 3.0.1
      PROJ: 7.2.1
  data dir: /home/user/envs/bp/lib/python3.6/site-packages/pyproj/proj_dir/share/proj
user_data_dir: /home/user/.local/share/proj

System:
    python: 3.6.9 (default, Jul  3 2019, 15:36:16)  [GCC 5.4.0 20160609]
executable: /home/user/envs/bp/bin/python3
   machine: Linux-4.15.0-107-generic-x86_64-with-Ubuntu-16.04-xenial

Python deps:
       pip: 20.0.2
setuptools: 46.1.3
    Cython: None
```

for pyproj==2.6.0:
```
pyproj info:
    pyproj: 2.6.0
      PROJ: 6.3.1
  data dir: /home/user/envs/bp/lib/python3.6/site-packages/pyproj/proj_dir/share/proj

System:
    python: 3.6.9 (default, Jul  3 2019, 15:36:16)  [GCC 5.4.0 20160609]
executable: /home/user/envs/bp/bin/python
   machine: Linux-4.15.0-107-generic-x86_64-with-Ubuntu-16.04-xenial

Python deps:
       pip: 20.0.2
setuptools: 46.1.3
    Cython: None
```

#### Installation method
```
pip3 install pyproj==3.0.1
pip3 install pyproj==2.6.0
```

**Comment by snowman2** [comment]
I wonder if using PROJ 8.1+ speeds things up: https://github.com/OSGeo/PROJ/pull/2738#issuecomment-855329515

**Comment by snowman2** [comment]
Looks like it has been resolved with PROJ 8.1+:
```bash
python -m timeit -s "import pyproj; crs = pyproj.CRS(4326)" "assert crs == crs"
```
```
100000 loops, best of 5: 3.38 usec per loop
```
Version information:
```
pyproj: 3.1.0
PROJ: 8.1.0
```
